salt/client/__init__.py

```diff
@@ -863,9 +882,9 @@ def get_iter_returns(
         # iterator for this job's return
         if self.opts['order_masters']:
             # If we are a MoM, we need to gather expected minions from downstreams masters.
-            ret_iter = self.get_returns_no_block(jid, tags_regex=['^syndic/.*/{0}'.format(jid)])
+            ret_iter = self.get_returns_short_block(jid, tags_regex=['^syndic/.*/{0}'.format(jid)], wait=timeout / 4)
         else:
-            ret_iter = self.get_returns_no_block(jid)
+            ret_iter = self.get_returns_short_block(jid, wait=timeout / 4)
```

Comment:
I wonder if the evaluated value is to low or to high depending on the timeout value in the test.   The believe the default value is 5.  In my testing I was using 30.    Can we log what the test is using.  Perhaps we need to set a min?

One other thing I was not clear on was whether the yield emedded in the wrapper method would work.  
Rather than have one wrap the other perhaps it should be separated out?
That should not really be a problem in this case though.